apps/front/components/TripMap.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <div>
+    <GoogleMap
+      :api-key="apiKey"
+      :center="center"
+      :street-view-control="false"
+      :zoom-control="false"
+      :fullscreen-control="false"
+      :map-type-control="false"
+      :zoom="7"
+      class="w-screen h-screen"
+    >
+      <Marker :options="markerStart" />
+      <Marker :options="markerEnd" />
+      <Marker
+        v-for="(route, index) in props.routes.slice(1, -1)"
+        :key="index"
+        :options="{
+          position: { lat: route.lat, lng: route.long },
+          icon: markerIcon,
+          label: { text: (index + 1).toString(), color: 'white' },
+          title: route.name,
+        }"
+      />
+      <Polyline :options="tripPath" />
+    </GoogleMap>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { GoogleMap, Marker, Polyline } from "vue3-google-map";
+import { decode } from '@mapbox/polyline';
+
+const { t } = useI18n();
+
+const runtimeConfig = useRuntimeConfig();
+const apiKey = `${runtimeConfig.public.GOOGLE_API_KEY}`;
+
+const props = defineProps({
+  routes: {
+    type: Object,
+    default: () => ({}),
+  },
+});
```

Comment:
```suggestion
const props = defineProps<{
  routes: {} // type the object
}>();
```

Please use typescript